software/teachee-desktop/src/sample_source/mod.rs

```diff
@@ -0,0 +1,91 @@
+use std::{marker::PhantomData, thread, time::Duration};
+
+use crate::storage::Storage;
+
+mod ft;
+mod sine;
+
+// flatten module structure
+pub use ft::FtSampleSource;
+pub use sine::SineSampleSource;
+
+pub type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+pub enum Channel {
+    VoltageA,
+    VoltageB,
+    CurrentA,
```

Comment:
This enum represents the channels available on the scope right? We only have one current channel.
Something more like this?
```
pub enum Channel {
    VoltageA,
    VoltageB,
    VoltageC,
    Current,
}
```